hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/RegionScanner.java

```diff
@@ -115,4 +116,6 @@
    */
   boolean nextRaw(List<Cell> result, ScannerContext scannerContext)
       throws IOException;
+
+  boolean isFullRegionScan();
```

Comment:
If we are doing this way - then I think it is better to have the default implementation in the interface returning 'false'. So that the downstreamers need not change this code. I think it is still better to have this check in RS (like a Util method)only rather than here because this will have an impact on projects like Phoenix. (because RegionScanner is exposed to CP).

Reply:
Agree.  After seeing this wanted to say abt def impl in interface (at least)..  But ya these methods in CP exposed interfaces make more work for phoenix kind of CPs which wraps the RegionScanner.  Exposing the user scan or its start/stop rows directly also not correct.  We cleaned all these stuff in 2+ versions.
Why cant we do the new metric accounting in lower layer of HRegion ? not in RS

Reply:
Moved metric accounting to HRegion layer now.